evaluator/evaluator/tasks/predict_task.py

```diff
@@ -88,42 +88,48 @@ def do_predict(
     headers = {"Authorization": f"Bearer {token}"}
     predictions: List[Prediction] = []
     start_time = datetime.datetime.now()
-    num_samples = 8  # len(dataset)
-    for i in range(num_samples):
-        logger.debug(f"Predicting sample {i + 1}/{num_samples}")
-        reference_data = dataset[i]
-
-        if dataset_metadata.skill_type == "extractive-qa":
-            query_request = {
-                "query": reference_data["question"],
-                "skill_args": {"context": reference_data["context"]},
-                "num_results": 1,
-            }
-        elif dataset_metadata.skill_type == "multiple-choice":
-            query_request = {
-                "query": reference_data["question"],
-                "skill_args": {"choices": reference_data["choices"]},
-                "num_results": 1,
-            }
-        else:
-            skill_type = dataset_metadata.skill_type
-            raise ValueError(
-                f"Predictions on '{skill_type}' datasets are currently not supported."
-            )
 
-        response = requests.post(
-            f"{base_url}/skill-manager/skill/{skill_id}/query",
-            headers=headers,
-            data=json.dumps(query_request),
+    if dataset_metadata["skill-type"] == "extractive-qa":
+        # 62c1ae1b536b1bb18ff91ce3 # squad
+        qa_type = "context"
+    elif dataset_metadata["skill-type"] == "multiple-choice":
+        # 62c1ae19536b1bb18ff91cde # commonsense_qa
+        qa_type = "choices"
+    else:
+        skill_type = dataset_metadata["skill-type"]
+        raise HTTPException(
+            400,
+            f"Predictions on '{skill_type}' datasets is currently not supported.",
         )
-        response.raise_for_status()
-        prediction_response = response.json()["predictions"][0]
 
+    query_request = {
+        "query": [
+            datapoint["question"] for datapoint in dataset if "question" in datapoint
+        ],
+        "skill_args": {
+            "context": [
+                datapoint[qa_type] for datapoint in dataset if qa_type in datapoint
```

Comment:
Hmm irgendwie fande ich den Code vorher lesbarer. Jetzt ist das irgendwie komisch. Erst schaust du das QA type an un den dann parst du den query request (aber abhängig vom qa type). Davor wurde alles in dem "if"-case abgefrühstückt...

Reply:
Ja. Finde zudem "_qa_type_" komisch gewählt. Weil "context" oder "choices" ist ja nicht der QA-Typ.

Reply:
Ich dachte mir ich es ändere es damit wir nicht 2 mal die gleiche Zeile Code haben, kann es aber gerne wieder zurückändern. Was ist euch lieber, neuer Variablenname oder das setzen des query_requests zurück in die if-else Abfrage reinverlagern?